test/test_torch.py

```diff
@@ -5353,6 +5353,120 @@ def add_neg_dim_tests():
+            (False, True, False),
+        )
+
+        with self.assertRaises(RuntimeError):
+            self._isclose_helper(tests, device, torch.bool, False)
+
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.uint8,
+            torch.int8, torch.int16, torch.int32, torch.int64)
+    def test_isclose_integer(self, device, dtype):
+        tests = (
+            (0, 0, True),
+            (0, 1, False),
+            (1, 0, False),
+        )
+
+        self._isclose_helper(tests, device, dtype, False)
+
+    # torch.close is not implemented for cpu half tensors
+    # see https://github.com/pytorch/pytorch/issues/36451
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.float16, torch.float32, torch.float64)
+    def test_isclose_float(self, device, dtype):
+        tests = (
+            (0, 0, True),
+            (0, -1, False),
+            (float('inf'), float('inf'), True),
+            (-float('inf'), float('inf'), False),
+            (float('inf'), float('nan'), False),
+            (float('nan'), float('nan'), False),
+            (0, float('nan'), False),
+            (1, 1, True),
+        )
+
+        if dtype is torch.half and self.device_type == 'cpu':
+            with self.assertRaises(RuntimeError):
+                self._isclose_helper(tests, device, dtype, False)
+        else:
+            self._isclose_helper(tests, device, dtype, False)
+
+        tests = (
+            (0, float('nan'), False),
+            (float('inf'), float('nan'), False),
+            (float('nan'), float('nan'), True),
+        )
+
+        if dtype is torch.half and self.device_type == 'cpu':
+            with self.assertRaises(RuntimeError):
+                self._isclose_helper(tests, device, dtype, True)
+        else:
+            self._isclose_helper(tests, device, dtype, True)
+
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.complex64, torch.complex128)
+    def test_isclose_complex(self, device, dtype):
+        tests = (
+            ((1 + 1j), complex(1, 1 + 1e-8), True),
+            ((0 + 1j), (1 + 1j), False),
+            ((1 + 1j), (1 + 0j), False),
+            ((1 + 1j), complex(1, float('nan')), False),
+            ((1 + 1j), complex(1, float('inf')), False),
+            ((float('inf') + 1j), complex(1, float('inf')), False),
+            ((-float('inf') + 1j), complex(1, float('inf')), False),
+            ((-float('inf') + 1j), complex(float('inf'), 1), False),
+            ((-float('inf') + 1j), (-float('inf') + 1j), True),
+            ((-float('inf') + 1j), complex(-float('inf'), 1 + 1e-4), False),
+            ((0 + 1j), complex(0, 1 + 1e-8), True),
+        )
+
+        self._isclose_helper(tests, device, dtype, False)
+
+        tests = (
+            ((1 + 1j), complex(1, float('nan')), False),
+            ((float('nan') + 1j), complex(1, float('nan')), False),
+            ((float('nan') + 1j), complex(1, 1), False),
```

Comment:
nit: maybe construct everything via `complex`? Here e.g. `complex` was not strictly necessary, as in some other places, but rather than deciding whether to use it or not, maybe easier to use it everywhere.

Reply:
Agreed.